ecs-cli/modules/config/readwriter.go

```diff
@@ -46,42 +41,79 @@ type ReadWriter interface {
 // compose-project-name-prefix = ecscompose-
 // compose-service-name-prefix =
 // cfn-stack-name-prefix = ecs-cli-
-
-type IniReadWriter struct {
+type INIReadWriter struct {
 	*Destination
 	cfg *ini.File
 }
 
-// NewReadWriter creates a new Parser object.
-func NewReadWriter() (*IniReadWriter, error) {
-	dest, err := newDefaultDestination()
-	if err != nil {
-		return nil, err
-	}
+// NewINIReadWriter creates a new Ini Parser object for the old ini configs
+func NewINIReadWriter(dest *Destination) (*INIReadWriter, error) {
 
-	iniCfg, err := newIniConfig(dest)
+	iniCfg, err := newINIConfig(dest)
 	if err != nil {
 		return nil, err
 	}
 
-	return &IniReadWriter{Destination: dest, cfg: iniCfg}, nil
+	return &INIReadWriter{Destination: dest, cfg: iniCfg}, nil
 }
 
 // GetConfig gets the ecs-cli config object from the config file.
-func (rdwr *IniReadWriter) GetConfig() (*CliConfig, error) {
-	to := new(CliConfig)
-	err := rdwr.cfg.MapTo(to)
+// map contains the keys that are present in the config file (maps string field name to string field value)
+// map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
+// cause certain keys to be mapped to maps of keys
+func (rdwr *INIReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
+	configMap := make(map[interface{}]interface{})
+	to := &CLIConfig{}
+
+	// read old ini formatted file
+	oldFormat := &iniCLIConfig{iniSectionKeys: new(iniSectionKeys)}
+	err := rdwr.cfg.MapTo(oldFormat)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return to, nil
+	// Create the configMap
+	if rdwr.IsKeyPresent(ecsSectionKey, "cluster") {
```

Comment:
Should we only check if the prefixes exist? The old format expect the other fields to be always present, so we can assume that the key always exists in this case.